### DISCUSSION TITLE
Improve price querry speed

### DISCUSSION THREAD

**Discussion by h4ckm1n-dev** [yuichiroaoki/poly-flashloan-bot#59] [Ideas]
Hello . i was talking with a friend (who is a dev ) of your bot, and he said you should query price from your own polygon node to have faster response. 

Maybe this could help so i share the idea

**Comment by bpevreall** [comment]
I set up a polygon node on AWS with a t2.2xLarge, unfortunatley the cost was too expensive before i worked out the kinks and tested properly. 

however 0.1s in most cases on amazons network infrastructure and alchemy RPC, which was much better than the 5s i received on my personal network.
i beleive this would work with your own node, though i dont have the means to provide results

**Comment by h4ckm1n-dev** [comment]
or maybe use a services like moralys

https://docs.moralis.io/introduction/readme
